welly/curve.py

```diff
@@ -286,92 +285,20 @@ def plot_2d(self, ax=None,
-        elif self.ndim == 2:
-            a = self[:, :width] if width < self.shape[1] else self
-        elif self.ndim == 3:
-            if 2 < self.shape[-1] < 5:
-                # Interpret as RGB or RGBA.
-                a = utils.normalize(np.copy(self))
-                cmap = None  # Actually doesn't matter.
-            else:
-                # Take first slice.
-                a = self[:, :width, 0] if width < self.shape[1] else self[..., 0]
-        else:
-            raise NotImplementedError("Can only handle up to 3 dimensions.")
-
-        # At this point, a is either a 2D array, or a 2D (rgb) array.
-        extent = [0, width or default, self.stop, self.start]
-        im = ax.imshow(a, cmap=cmap, extent=extent, aspect='auto')
-
-        if curve:
-            paths = ax.fill_betweenx(self.basis, self, np.nanmin(self),
-                                     facecolor='none',
-                                     **kwargs,
-                                     )
+        return plot_2d_curve(curve=self,
+                             ax=ax,
+                             width=width,
+                             aspect=aspect,
+                             cmap=cmap,
+                             plot_curve=curve,
+                             ticks=ticks,
+                             return_fig=return_fig,
+                             **kwargs)
 
-            # Make the 'fill' mask and clip the background image with it.
-            patch = PathPatch(paths._paths[0], visible=False)
-            ax.add_artist(patch)
-            im.set_clip_path(patch)
-
-        ax.set_xticks([])
-
-        # Rely on interval order.
-        lower, upper = self.stop, self.start
-        rng = abs(upper - lower)
-
-        ax.set_ylim([lower, upper])
-
-        # Make sure ticks is a tuple.
-        try:
-            ticks = tuple(ticks)
-        except TypeError:
-            ticks = (1, ticks)
-
-        # Avoid MAXTICKS error.
-        while rng/ticks[0] > 250:
-            mi, ma = 10*ticks[0], ticks[1]
-            if ma <= mi:
-                ma = 10 * mi
-            ticks = (mi, ma)
-
-        # Carry on plotting...
-        minorLocator = mpl.ticker.MultipleLocator(ticks[0])
-        ax.yaxis.set_minor_locator(minorLocator)
-
-        majorLocator = mpl.ticker.MultipleLocator(ticks[1])
-        majorFormatter = mpl.ticker.FormatStrFormatter('%d')
-        ax.yaxis.set_major_locator(majorLocator)
-        ax.yaxis.set_major_formatter(majorFormatter)
-
-        ax.yaxis.set_ticks_position('left')
-        ax.get_yaxis().set_tick_params(which='both', direction='out')
-
-        if return_ax:
-            return ax
-        elif return_fig:
-            plt.tight_layout()
-            return fig
-        else:
-            return None
-
-    def plot(self, ax=None, legend=None, return_fig=False, alias=None, **kwargs):
+    def plot(self, ax=None, legend=None, return_fig=False, **kwargs):
```

Comment:
was `alias` used at all here?
If it was, passing the `kwargs` on might cause an error.

Reply:
`alias` was not used